test/extended/builds/pipeline.go

```diff
@@ -776,6 +778,63 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
 				o.Expect(err).NotTo(o.HaveOccurred())
 			})
 
+			g.By("Pipeline with env vars and git repo source")
+
+			g.It("should propagate env vars to bc", func() {
+				g.By(fmt.Sprintf("creating git repo %v", envVarsPipelineGitRepo))
+				jf := `node() {\necho "FOO1 is ${env.FOO1}"\necho"FOO2is${env.FOO2}"\necho"FOO3is${env.FOO3}"\necho"FOO4is${env.FOO4}"}`
+				repo, err := exutil.NewGitRepo(envVarsPipelineGitRepo)
+				defer repo.Cleanup()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				err = repo.AddAndCommit("Jenkinsfile", jf)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				// instantiate the bc
+				g.By(fmt.Sprintf("calling oc new-app %q --strategy=pipeline --build-env=FOO1=BAR1", envVarsPipelineGitRepo))
+				err = oc.Run("new-app").Args(envVarsPipelineGitRepo, "--strategy=pipeline", "--build-env=FOO1=BAR1").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				bc, err := oc.BuildClient().Build().BuildConfigs(oc.Namespace()).Get(envVarsPipelineGitRepoBuildConfig, metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				envs := bc.Spec.Strategy.JenkinsPipelineStrategy.Env
+				o.Expect(len(envs)).To(o.Equal(1))
+				o.Expect(envs[0].Name).To(o.Equal("FOO1"))
+				o.Expect(envs[0].Value).To(o.Equal("BAR1"))
+
+				//TODO: start the build and validate envs as well
```

Comment:
this shouldn't be necessary, we ought to already have a test somewhere that covers "pipeline buildconfig w/ env vars" behavior.  @gabemontero should be able to point to it.

I think it should be sufficient to confirm the buildconfig that is generated, has the env vars.

Reply:
yes, we do, I located this test right below the pipeline buildconfig but I thought since the issue complained about both parts, we may want to test it all through here as well.